src/ComputerVirtualMachine.php

```diff
@@ -434,6 +434,19 @@ public static function findVirtualMachine($fields = [])
             $result = $iterator->current();
             return $result['id'];
         } else if (count($iterator) > 1) {
+            $iterator = $DB->request([
+                'SELECT' => ['id'],
+                'FROM'   => 'glpi_computers',
+                'WHERE'  => [
+                    'RAW' => [
+                        'LOWER(uuid)'  => $fields['uuid']
+                    ]
+                ]
+            ]);
+            if (count($iterator) == 1) {
+                $result = $iterator->current();
+                return $result['id'];
+            }
```

Comment:
The first request already returns corresponding matches. If we want to return the corresponding computer if there is one and only one exact uuid match, then it should be done without any extra request.

Still, it is just a technical remark. I do not know if this is a valid behaviour.

Reply:
This is a technical remark but isn't a valid behavior. It was designed to use the function ``getUUIDRestrictCriteria`` because of a bug on DMIDecode, as UUID can be wrongly calculated and to find a VM it need to be sorted as it maybe stored with another UUID. But if there's multiple UUID, to find the one of ESX host, we can use only the stored UUID on DB, which returns the correct VM and avoid the error on the screenshot.

Reply:
`getUUIDRestrictCriteria` will already permit to fetch the row that have an exact match with the providen UUID. Adding the `uuid` field to the selected fields (`'SELECT' => 'id'` -> `'SELECT' => ['id', 'uuid']`) would premit to filter the result to see if we found an exact match.